pallets/proofs-dealer/src/lib.rs

```diff
@@ -306,6 +305,13 @@ pub mod pallet {
 
         /// A slashable provider was found.
         SlashableProvider { provider: ProviderIdFor<T> },
+
+        /// A Provider's challenge cycle was initialised.
+        NewChallengeCycleInitialised {
+            current_tick: BlockNumberFor<T>,
+            provider: ProviderIdFor<T>,
+            maybe_provider_account: Option<T::AccountId>,
```

Comment:
It's because the proofs dealer pallet doesn't know that it is actually impossible for a provider to exist without an account. As far as it is concerned, it queries the account for a provider, and that CAN return None.